src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -228,6 +294,16 @@ public virtual double NextDouble()
             return Sample();
         }
 
+        /*==================================NextSingle==================================
+        **Returns: A float [0..1)
+        **Arguments: None
+        **Exceptions: None
+        ==============================================================================*/
+        public virtual float NextSingle()
+        {
+            return (float) Sample();
```

Comment:
The conversion from `double` to `float` is going to round unrepresentable `double` values to representable `float` ones - and it's not clear that the distribution of that is going to be even.  Is there a reference that can be added as a comment as to whether this is safe?